runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java

```diff
@@ -654,8 +637,22 @@ private void onStreamData(
 
             initialSeqDelta = connection.initialSeq;
 
+            if (requestBytes == 0)
+            {
+                nextRequestId++;
+
+                final DirectBuffer buffer = payload.buffer();
+                final int offset = payload.offset();
+                final int limit = payload.limit();
+
+                RequestHeaderFW requestHeader = requestHeaderRO.wrap(buffer, offset, limit);
+                requestBytes = requestHeader.length() + KAFKA_FRAME_LENGTH_FIELD_OFFSET;
+            }
+
+            requestBytes -= payload.sizeof();
```

Comment:
Strictly speaking a stream _could_ send multiple requests in the same `DATA` frame.
Here we are assuming that they don't with the assert.